operations/raft-storage/local/cluster.sh

```diff
@@ -0,0 +1,513 @@
+#!/bin/bash
+
+set -e
+
+DEMO_HOME=$(pwd)
+script_name=`basename "$0"`
+
+# Create a helper function to address the first vault node
+function vault_1() {
+    (export VAULT_ADDR=http://127.0.0.1:8200 && vault $@)
+}
+
+# Create a helper function to address the second vault node
+function vault_2() {
+    (export VAULT_ADDR=http://127.0.0.2:8200 && vault $@)
+}
+
+# Create a helper function to address the third vault node
+function vault_3() {
+    (export VAULT_ADDR=http://127.0.0.3:8200 && vault $@)
+}
+
+# Create a helper function to address the fourth vault node
+function vault_4() {
+    (export VAULT_ADDR=http://127.0.0.4:8200 && vault $@)
+}
+
+function stop() {
+  service_count=$(pgrep -f $(pwd)/config | wc -l | tr -d '[:space:]')
+
+  printf "\n%s" \
+    "Found $service_count Vault services" \
+    ""
+
+  if [ $service_count != "0" ] ; then
+    printf "\n%s" \
+      "Stopping $service_count Vault services" \
+      ""
+
+    pkill -f $(pwd)/config
+  fi
+}
+
+function loopback_exists_at_address() {
+  echo $(ifconfig lo0 | grep $1 || true) | tr -d '[:space:]'
+}
+
+function clean() {
+
+  printf "\n%s" \
+    "Cleaing up the HA cluster. Removing:" \
+    " - local loopback address for [vault_2], [vault_3], and [vault_4[" \
+    " - configuration files" \
+    " - raft storage directory" \
+    " - log files" \
+    " - unseal / recovery keys" \
+    ""
+
+  for loopback_address in "127.0.0.2" "127.0.0.3" "127.0.0.4" ; do
+    loopback_exists=$(loopback_exists_at_address $loopback_address)
+
+    if [[ $loopback_exists != "" ]] ; then
+      printf "\n%s" \
+        "Removing local loopback address: $loopback_address (sudo required)" \
+        ""
+
+      sudo ifconfig lo0 -alias $loopback_address
+    fi
+  done
+
+  for config_file in $DEMO_HOME/config-vault_1.hcl $DEMO_HOME/config-vault_2.hcl $DEMO_HOME/config-vault_3.hcl $DEMO_HOME/config-vault_4.hcl ; do
+    if [[ -f "$config_file" ]] ; then
+      printf "\n%s" \
+        "Removing configuration file $config_file"
+
+      rm $config_file
+    fi
+  done
+
+  for raft_storage in $DEMO_HOME/raft-vault_2 $DEMO_HOME/raft-vault_3 $DEMO_HOME/raft-vault_4 ; do
+    if [[ -d "$raft_storage" ]] ; then
+    printf "\n%s" \
+        "Removing raft storage file $raft_storage"
+
+      rm -rf $raft_storage
+    fi
+  done
+
+  for key_file in $DEMO_HOME/unseal_key-vault_1 $DEMO_HOME/recovery_key-vault_2 ; do
+    if [[ -f "$key_file" ]] ; then
+      printf "\n%s" \
+        "Removing key $key_file"
+
+      rm $key_file
+    fi
+  done
+
+
+  for vault_log in $DEMO_HOME/vault_1.log $DEMO_HOME/vault_2.log $DEMO_HOME/vault_3.log $DEMO_HOME/vault_4.log ; do
+    if [[ -f "$vault_log" ]] ; then
+      printf "\n%s" \
+        "Removing log file $vault_log"
+
+      rm $vault_log
+    fi
+  done
+
```

Comment:
Remove the `root_token-vault_1` and `root_token-vault_2` as well for the completeness.  

Anyway, this `clean` command is nice!